tests/test_nodes.cpp

```diff
@@ -150,3 +150,37 @@ TEST_CASE("NodePopulationSelectAll", "[base]") {
     NodePopulation population("./data/nodes1.h5", "", "nodes-A");
     CHECK(population.selectAll().flatSize() == 6);
 }
+
+TEST_CASE("NodePopulationmatchAttributeValues", "[base]") {
+    NodePopulation population("./data/nodes1.h5", "", "nodes-A");
+
+    SECTION("String") {
+        auto sel = population.matchAttributeValues<std::string>("attr-Z", std::string("bb"));
+        CHECK(sel.flatSize() == 1);
+        CHECK(Selection::fromValues({1}) == sel);
+    }
+
+    SECTION("Enumeration") {
+        auto sel = population.matchAttributeValues("E-mapping-good", std::string("C"));
+        CHECK(sel.flatSize() == 4);
+        CHECK(Selection::fromValues({0, 2, 4, 5}) == sel);
+    }
+}
+
+TEMPLATE_TEST_CASE("NodePopulationmatchAttributeValues",
+                   "Numeric",
+                   float,
+                   double,
+                   int8_t,
+                   uint8_t,
+                   int16_t,
+                   uint16_t,
+                   int32_t,
+                   uint32_t,
+                   int64_t,
+                   uint64_t) {
+    NodePopulation population("./data/nodes1.h5", "", "nodes-A");
+    auto sel = population.matchAttributeValues<TestType>("attr-Y", 23);
```

Comment:
Github is messing with my view.

But the data here should be generated with:
```
    attrs.create_dataset('attr-Y', data=np.arange(21., 27., 0.1), dtype=np.int64)
```
For the float/double test-cases.

Reply:
`dtype=np.int64`?

Reply:
ah, yes, that was left as an exercise for the test implementation. But now that you mention in, line 25 there reads:
```
    attrs.create_dataset('attr-Y', data=np.arange(21., 27.), dtype=np.int64)
```

Should be fixed, too?

Reply:
I'm not following. `data=np.arange(21., 27., 0.1)` would create more entries than there are supposed to be.  This is why I had to change `E-mapping-good` above, so that it has six elements.  I have fixed the `21.` issue.

Reply:
It doesn't have the same range. Basically making sure that type resolution works by creating a bunch of values that would get truncated to the same integer.